src/index/ivf/ivf_config.h

```diff
@@ -95,6 +110,16 @@ class ScannConfig : public IvfFlatConfig {
     Status
     CheckAndAdjust(PARAM_TYPE param_type, std::string* err_msg) override {
         switch (param_type) {
+            case PARAM_TYPE::TRAIN: {
+                // TODO: handle odd dim with scann
+                if (dim.has_value()) {
+                    int vec_dim = dim.value();
+                    if (vec_dim % 2 != 0) {
+                        *err_msg = "dimension must be able to be divided by 2, dimension:" + std::to_string(vec_dim);
+                        return Status::invalid_args;
```

Comment:
In Faiss, PQ requires dim % M == 0, see https://github.com/zilliztech/knowhere/blob/main/thirdparty/faiss/faiss/impl/ProductQuantizer.cpp#L60-L62; previously, there were no issues because Milvus enforced this constraint (https://github.com/milvus-io/milvus/blob/master/pkg/util/indexparamcheck/scann_checker.go#L33-L35).

Reply:
I didn't mean the PQ one but the `vec_dim % 2!=0`

Reply:
It's the same; SCANN also uses PQ and requires that dim be divisible by 2